include/openPMD/WriteIterations.hpp

```diff
@@ -50,6 +50,12 @@ class WriteIterations : private Container< Iteration, uint64_t >
 
 private:
     using iterations_t = Container< Iteration, uint64_t >;
+    
+public:
+    using key_type = typename iterations_t::key_type;
+    using mapped_type = typename iterations_t::mapped_type;
```

Comment:
Hah, needed in pybind11:
```suggestion
    using mapped_type = typename iterations_t::mapped_type;
    using value_type = typename iterations_t::value_type;
```